avro/src/decode.rs

```diff
@@ -123,61 +121,18 @@ pub(crate) fn decode_internal<R: Read, S: Borrow<Schema>>(
             }
         }
         Schema::Uuid => {
-            let len = decode_len(reader)?;
-            let mut bytes = vec![0u8; len];
-            reader
-                .read_exact(&mut bytes)
-                .map_err(Details::ReadIntoBuf)?;
-
-            // use a Vec to be able re-read the bytes more than once if needed
-            let mut reader = Vec::with_capacity(len + 1);
-            encode_long(len as i64, &mut reader)?;
-            reader.extend_from_slice(&bytes);
-
-            let decode_from_string = |reader| match decode_internal(
-                &Schema::String,
-                names,
-                enclosing_namespace,
-                reader,
-            )? {
-                Value::String(ref s) => {
-                    Uuid::from_str(s).map_err(|e| Details::ConvertStrToUuid(e).into())
-                }
-                value => Err(Error::new(Details::GetUuidFromStringValue(value))),
+            let Value::Bytes(bytes) =
+                decode_internal(&Schema::Bytes, names, enclosing_namespace, reader)?
+            else {
+                // Calling decode_internal with Schema::Bytes can only return a Value::Bytes or an error
```

Comment:
do we need the `else` at all ?

Reply:
Yes, otherwise the compiler will throw an error as it can't prove that this is the only option